packages/Python/lldbsuite/test/lang/swift/expression/exclusivity_suppression/TestExclusivitySuppression.py

```diff
@@ -0,0 +1,145 @@
+    (exit_status, output) = commands.getstatusoutput(command)
+    return exit_status
+
+class TestExclusivitySuppression(TestBase):
+
+    mydir = TestBase.compute_mydir(__file__)
+
+    # Test that we can evaluate w.s.i at Breakpoint 1 without triggering
+    # a failure due to exclusivity
+    @decorators.swiftTest
+    def test_basic_exclusivity_suppression(self):
+        """Test that exclusively owned values can still be accessed"""
+
+        self.buildAll()
+
+        target = self.create_target()
+
+        # Set the breakpoints
+        bp1 = target.BreakpointCreateBySourceRegex('Breakpoint 1',
+                                                   self.main_source_spec)
+        self.assertTrue(bp1.GetNumLocations() > 0, VALID_BREAKPOINT)
+
+        # Launch the process, and do not stop at the entry point.
+        process = target.LaunchSimple(None, None, os.getcwd())
+
+        self.assertTrue(process, PROCESS_IS_VALID)
+
+        # Frame #0 should be at our breakpoint.
+        threads = lldbutil.get_threads_stopped_at_breakpoint(process, bp1)
+
+        self.assertTrue(len(threads) == 1)
+        thread = threads[0]
+        frame = thread.frames[0]
+        self.assertTrue(frame, "Frame 0 is valid.")
+
+        self.check_expression(frame, "w.s.i", "8", use_summary=False)
+
+    # Test that we properly handle nested expression evaluations by:
+    # (1) Breaking at breakpoint 1
+    # (2) Running 'expr get()' (which will hit breakpoint 2)
+    # (3) Evaluating w.s.i at breakpoint 2 (this is a nested evaluation)
+    # (4) Contininging the evaluation of 'expr get()' to return to bp 1
+    # (5) Evaluating w.s.i again to check that finishing the nested expression
+    #     did not prematurely re-enable exclusivity checks.
+    @decorators.swiftTest
+    def test_exclusivity_suppression_for_concurrent_expressions(self):
+        """Test that exclusively owned values can still be accessed"""
+        self.buildAll()
+
+        target = self.create_target()
+
+        bp1 = target.BreakpointCreateBySourceRegex('Breakpoint 1',
+                                                   self.main_source_spec)
+        self.assertTrue(bp1.GetNumLocations() > 0, VALID_BREAKPOINT)
+
+        bp2 = target.BreakpointCreateBySourceRegex('Breakpoint 2',
+                                                   self.main_source_spec)
+        self.assertTrue(bp2.GetNumLocations() > 0, VALID_BREAKPOINT)
+
+        # Launch the process, and do not stop at the entry point.
+        process = target.LaunchSimple(None, None, os.getcwd())
+
+        self.assertTrue(process, PROCESS_IS_VALID)
+
+        # Frame #0 should be at our breakpoint.
+        threads = lldbutil.get_threads_stopped_at_breakpoint(process, bp1)
+
+        self.assertTrue(len(threads) == 1)
+        thread = threads[0]
+        frame = thread.frames[0]
+        self.assertTrue(frame, "Frame 0 is valid.")
+
+        self.check_expression(frame, "w.s.i", "8", use_summary=False)
+
```

Comment:
Oops - this was meant to be left out of this commit and added in a follow-up.

I have a "complete" version of this test but it fails due to unexpected behavior in SBProcess::continue - I'll see if I can get that sorted out along with the other issues you've flaged and post the full test.